cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -282,22 +284,74 @@ private void copyRecursive(Path source, Path target, FileCopyMode mode) throws I
     }
   }
 
+  @Override
+  public void makeSymlinkRelative(Path link) {
+
+    makeSymlinkRelative(link, false);
+  }
+
+  @Override
+  public void makeSymlinkRelative(Path link, boolean followTarget) {
+
+    if (!Files.isSymbolicLink(link)) {
+      throw new IllegalStateException(
+          "Can't call makeSymlinkRelative on " + link + " since it is not a symbolic link.");
+    }
+    Path linkTarget = null;
```

Comment:
`= null` is not needed as Object are initialized as null so `Path linkTarget` should work aswell

Reply:
Only member variables are automatically initialized as null (and array items) but not local variables.
However, as the variable gets initialized in 302 and in case of an IOException we exit with a RuntimeException the compiler is smart enough to understand it is initialized in 308 so @moritzLanger is right that the assignment is not needed.